authority/provisioner/aws.go

```diff
@@ -271,7 +272,7 @@ func (p *AWS) Init(config Config) (err error) {
 func (p *AWS) AuthorizeSign(ctx context.Context, token string) ([]SignOption, error) {
 	payload, err := p.authorizeToken(token)
 	if err != nil {
-		return nil, err
+		return nil, errs.Wrap(http.StatusInternalServerError, err, "authorizeSign")
```

Comment:
It looks like the errors from `authorizeToken` already have a custom status code. Would this override the specific code returned in `authorizeToken`

Reply:
No, this wouldn't override (you can see that in the unit tests). It will only replace the status code if the returned error was not a StatusCoder. Basically, it's a catchall to make sure we always return a StatusCoder.

Reply:
Ok, I've looked at the implementation of `errs.Wrap`, and `errs.StatusCodeError`, called from `Wrap` and it looked to me that it was overwriting it, I didn't notice that this calls to `errs.New` that puts the initial error code instead.

Reply:
Looking again I think the error displayed to the user might be in some cases `The certificate authority encountered an Internal Server Error.` even that the status was unauthorized.

Reply:
You were right. I believe I fixed this now.